File: src/main/java/com/procurement/submission/controller/PeriodController.java
package com.procurement.submission.controller;

import com.procurement.submission.model.dto.bpe.ResponseDto;
import com.procurement.submission.service.PeriodService;
import java.time.LocalDateTime;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(path = "/period")
public class PeriodController {

    private PeriodService periodService;

    public PeriodController(final PeriodService periodService) {
        this.periodService = periodService;
    }

    @PostMapping("/save")
    public ResponseEntity<ResponseDto> savePeriod(@RequestParam("identifier") final String cpId,
                                                  @RequestParam("stage") final String stage,
                                                  @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                                                  @RequestParam("startDate") final LocalDateTime startDate,
                                                  @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                                                  @RequestParam("endDate") final LocalDateTime endDate) {
        return new ResponseEntity<>(periodService.savePeriod(cpId, stage, startDate, endDate), HttpStatus.CREATED);
    }

    @PostMapping("/new")
    public ResponseEntity<ResponseDto> saveNewPeriod(@RequestParam("identifier") final String cpId,
                                                     @RequestParam("stage") final String stage,
                                                     @RequestParam("country") final String country,
                                                     @RequestParam("pmd") final String pmd,
                                                     @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                                                     @RequestParam("startDate") final LocalDateTime startDate) {
        return new ResponseEntity<>(periodService.saveNewPeriod(cpId, stage, country, pmd, startDate), HttpStatus
                .CREATED);
    }

    @PostMapping("/validation")
    public ResponseEntity<ResponseDto> periodValidation(@RequestParam("country") final String country,
                                                        @RequestParam("pmd") final String pmd,
                                                        @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                                                        @RequestParam("startDate") final LocalDateTime startDate,
                                                        @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                                                        @RequestParam("endDate") final LocalDateTime endDate) {
        return new ResponseEntity<>(periodService.periodValidation(country, pmd, startDate, endDate),
                HttpStatus.OK);
    }

    @PostMapping("/check")
    public ResponseEntity<ResponseDto> checkPeriod(@RequestParam("identifier") final String cpId,
                                                   @RequestParam("stage") final String stage,
                                                   @RequestParam("country") final String country,
                                                   @RequestParam("pmd") final String pmd,
                                                   @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                                                   @RequestParam("startDate") final LocalDateTime startDate,
                                                   @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                                                   @RequestParam("endDate") final LocalDateTime endDate) {
        return new ResponseEntity<>(periodService.checkPeriod(cpId, country, pmd, stage, startDate, endDate),
                HttpStatus.OK);
    }
}

File: src/main/kotlin/com/procurement/submission/controller/PeriodController.kt
package com.procurement.submission.controller

import com.procurement.submission.model.dto.bpe.ResponseDto
import com.procurement.submission.service.PeriodService
import org.springframework.format.annotation.DateTimeFormat
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.time.LocalDateTime

@RestController
@RequestMapping(path = ["/period"])
class PeriodController(private val periodService: PeriodService) {


    @GetMapping
    fun getPeriod(@RequestParam("cpid") cpId: String,
                  @RequestParam("stage") stage: String): ResponseEntity<ResponseDto> {
        return ResponseEntity(
                periodService.getPeriod(cpId = cpId, stage = stage),
                HttpStatus.CREATED)
    }

    @PostMapping("/save")
    fun savePeriod(@RequestParam("cpid") cpId: String,
                   @RequestParam("stage") stage: String,
                   @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                   @RequestParam("startDate") startDate: LocalDateTime,
                   @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                   @RequestParam("endDate") endDate: LocalDateTime): ResponseEntity<ResponseDto> {
        return ResponseEntity(
                periodService.savePeriod(
                        cpId = cpId,
                        stage = stage,
                        startDate = startDate,
                        endDate = endDate),
                HttpStatus.CREATED)
    }

    @PostMapping("/new")
    fun saveNewPeriod(@RequestParam("cpid") cpId: String,
                      @RequestParam("stage") stage: String,
                      @RequestParam("country") country: String,
                      @RequestParam("pmd") pmd: String,
                      @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                      @RequestParam("startDate") startDate: LocalDateTime): ResponseEntity<ResponseDto> {
        return ResponseEntity(
                periodService.saveNewPeriod(
                        cpId = cpId,
                        stage = stage,
                        country = country,
                        pmd = pmd,
                        startDate = startDate),
                HttpStatus.CREATED)
    }

    @PostMapping("/validation")
    fun periodValidation(@RequestParam("country") country: String,
                         @RequestParam("pmd") pmd: String,
                         @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                         @RequestParam("startDate") startDate: LocalDateTime,
                         @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                         @RequestParam("endDate") endDate: LocalDateTime): ResponseEntity<ResponseDto> {
        return ResponseEntity(
                periodService.periodValidation(
                        country = country,
                        pmd = pmd,
                        startDate = startDate,
                        endDate = endDate),
                HttpStatus.OK)
    }

    @PostMapping("/check")
    fun checkPeriod(@RequestParam("cpid") cpId: String,
                    @RequestParam("stage") stage: String,
                    @RequestParam("country") country: String,
                    @RequestParam("pmd") pmd: String,
                    @RequestParam("operationType") operationType: String,
                    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                    @RequestParam("startDate") startDate: LocalDateTime,
                    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
                    @RequestParam("endDate") endDate: LocalDateTime): ResponseEntity<ResponseDto> {
        return ResponseEntity(
                periodService.checkPeriod(
                        cpId = cpId,
                        country = country,
                        pmd = pmd,
                        operationType = operationType,
                        stage = stage,
                        requestDate = startDate,
                        endDateReq = endDate),
                HttpStatus.OK)
    }
}
